presto-hive/src/main/java/com/facebook/presto/hive/HiveStagingFileCommitter.java

```diff
@@ -67,13 +69,22 @@ public void commitFiles(ConnectorSession session, String schemaName, String tabl
             }
         }
 
-        ListenableFuture<?> listenableFutureAggregate = whenAllSucceed(commitFutures).call(() -> null, directExecutor());
-        try {
-            getFutureValue(listenableFutureAggregate, PrestoException.class);
-        }
-        catch (RuntimeException e) {
-            listenableFutureAggregate.cancel(true);
-            throw e;
-        }
+        ListenableFuture<?> result = whenAllSucceed(commitFutures).call(() -> null, directExecutor());
+        ListenableFuture<?> ioExceptionHandledResult = catching(
+                result,
+                IOException.class,
+                e -> {
+                    result.cancel(true);
+                    return new PrestoException(HIVE_FILESYSTEM_ERROR, "Error moving data files to final destination", e);
+                },
+                directExecutor());
+        return catching(
```

Comment:
Per offline discussion, we don't seem to need two `catching`?

Reply:
Done